### DISCUSSION TITLE
CSS formating for the writer area?

### DISCUSSION THREAD

**Discussion by glennhefley** [vkbo/novelWriter#1500] [General]
Hi, 

Love the program. I've been working as a ghost writer for many years. I was a big fan of Scrivener but the new version is not to my liking. Yours is wonderful in that it is fast, and light, and not stuffed with things I don't need. 

I would like to alter the spacing between lines within paragraphs however. I was thinking that there might be a CSS file some place I could edit for that purpose, but haven't found one. Any ideas?

**Comment by vkbo** [comment]
The topic of line height has been discussed a fair bit. There is no setting that allows changing it for the [QTextEdit](https://doc.qt.io/qt-5/qtextedit.html) widget that novelWriter uses. It is really frustrating that they have not exposed this setting, but they haven't. Setting it in CSS does nothing.

The only way I've found to do this is to apply the line height to each paragraph at a time, which works, but is incredibly slow when doing it from Python, like novelWriter does. See #654.

There's also a thread on StackOverflow on this, and my own solution is [posted there](https://stackoverflow.com/a/66091502/5825851). It works, but cannot be used in practice. The solution works just fine when done from C++, which is what the Qt developers focus on, so they seem to have ignored the issue as far as Python is concerned. It's been an issue for over a decade.

**Reply by glennhefley** [reply]
thank you for the response. If this turns out to be the only trouble I have with the software, that would be wonderful, and acceptable. So far it's doing just fine. thanks!
